pkg/apis/core/v1beta1/constants/types_constants.go

```diff
@@ -570,10 +570,16 @@ const (
 	// delays will not recompute it.
 	AnnotationShootCloudConfigExecutionMaxDelaySeconds = "shoot.gardener.cloud/cloud-config-execution-max-delay-seconds"
 	// AnnotationNodeLocalDNS enables a per node dns cache on the shoot cluster.
+	// Deprecated: This annotation is deprecated and will be removed in a future version.
+	// Use field `.spec.systemComponents.nodeLocalDNS.enabled` in Shoot instead.
 	AnnotationNodeLocalDNS = "alpha.featuregates.shoot.gardener.cloud/node-local-dns"
 	// AnnotationNodeLocalDNSForceTcpToClusterDns enforces upgrade to tcp connections for communication between node local and cluster dns.
+	// Deprecated: This annotation is deprecated and will be removed in a future version.
+	// Use field `.spec.systemComponents.nodeLocalDNSforceTCPToClusterDNS` in Shoot instead.
 	AnnotationNodeLocalDNSForceTcpToClusterDns = "alpha.featuregates.shoot.gardener.cloud/node-local-dns-force-tcp-to-cluster-dns"
 	// AnnotationNodeLocalDNSForceTcpToUpstreamDns enforces upgrade to tcp connections for communication between node local and upstream dns.
+	// Deprecated: This annotation is deprecated and will be removed in a future version.
+	// Use field `.spec.systemComponents.nodeLocalDNSforceTCPToUpstreamDNS` in Shoot instead.
 	AnnotationNodeLocalDNSForceTcpToUpstreamDns = "alpha.featuregates.shoot.gardener.cloud/node-local-dns-force-tcp-to-upstream-dns"
 	// AnnotationCoreDNSRewritingDisabled disables core dns query rewriting even if the corresponding feature gate is enabled.
 	AnnotationCoreDNSRewritingDisabled = "alpha.featuregates.shoot.gardener.cloud/core-dns-rewriting-disabled"
```

Comment:
Should we also deprecate this one? cc @ScheererJ @DockToFuture @axel7born

Reply:
No, `AnnotationCoreDNSRewritingDisabled` should not be deprecated, yet. Let's first promote the corresponding feature gate.